### PR TITLE
fix: [CI-4357]: TI Harness Hierarchy example

### PR DESCRIPTION
**Pull request by jamie-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)

### CODE REVIEW

**Comment by jamie-harness** [comment]
trigger messagecheck

File: 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 import static org.assertj.core.api.Assertions.assertThatCode;
 
+import io.harness.agent.sdk.HarnessHierarchy;
```
Comment (hemanthmantri) [inline 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java:14]: There's an existing io.harness.annotations folder. It feels natural to keep all annotations there. Isn't it? Also let's not keep agent in the name - exposing internals of working of TI which can change in the future to no use instrumentation agent.

File: 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 import static org.assertj.core.api.Assertions.assertThatCode;
 
+import io.harness.agent.sdk.HarnessHierarchy;
```
Comment (hemanthmantri) [inline 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java:14]: How about io.harness.annotations.tests.HarnessHierarchy ?

File: 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java [hunk 3]
```diff
@@ -32,6 +33,7 @@ public class CiBeansComponentTest extends CiBeansTestBase {
   @Test
   @Owner(developers = AMAN)
   @Category(UnitTests.class)
+  @HarnessHierarchy(TestExecution.class)
```
Comment (hemanthmantri) [inline 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java:36]: Please add a comment explaining what this is doing. In the example please also indicate how to specify a list of these dependent classes.

File: 980-commons/src/main/java/io/harness/morphia/MorphiaModule.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
 import static io.harness.govern.IgnoreThrowable.ignoredOnPurpose;
 import static io.harness.morphia.MorphiaRegistrar.putClass;
 
-import io.harness.annotations.ti.HarnessTrace;
```
Comment (hemanthmantri) [inline 980-commons/src/main/java/io/harness/morphia/MorphiaModule.java:13]: This seems better than agent.sdk
May be io.harness.annotations.tests.HarnessTrace

File: 999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.agent.sdk;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
```
Comment (hemanthmantri) [inline 999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java:14]: Document the usage in comments.

**Review by hemanthmantri — COMMENTED** [review]
A few nits mostly around package structure and documentation. Thanks for adding these annotations.

File: 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java (hunk 1 above)
Comment (jamie-harness) [reply 325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java:14]: This has to match what's defined in the agent ini file. The previous Tracer annotation does not work.

File: 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java (hunk 2 above)
Comment (jamie-harness) [reply 330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java:14]: No this has to match agent's .ini file definition.

File: 999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java (hunk 5 above)
Comment (jamie-harness) [reply 999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java:14]: Let's do that in the doc. Will need a few more lines than comment

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: 10 file(s) to review. 1. unresolved discussions. 

___
*[`325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java` line 14 at r1](https://reviewable.io/reviews/harness/harness-core/32737#-N19c8jT0CYaHrIdDxz6-r1-14:-N1Ezz5l62hnl3af52Fc:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/ea5ca62a437aa570476e29659bb25cdf846a2b95/325-sto-beans/src/test/java/io/harness/STOBeansComponentTest.java#L14)):*
<details><summary><i>Previously, jamie-harness (Jamie Li) wrote…</i></summary><blockquote>

This has to match what's defined in the agent ini file. The previous Tracer annotation does not work.
</blockquote></details>

OK
___
*[`330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java` line 14 at r1](https://reviewable.io/reviews/harness/harness-core/32737#-N19c8jT0CYaHrIdDxz8-r1-14:-N1F-1ouEmL6Q4OLqr0i:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/ea5ca62a437aa570476e29659bb25cdf846a2b95/330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java#L14)):*
<details><summary><i>Previously, jamie-harness (Jamie Li) wrote…</i></summary><blockquote>

No this has to match agent's .ini file definition.
</blockquote></details>

OK

**Review by jamie-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: 10 file(s) to review. 2. unresolved discussions. 

___
*[`330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java` line 36 at r1](https://reviewable.io/reviews/harness/harness-core/32737#-N19c8jT0CYaHrIdDxz8-r1-36:-N1F4PrZ-9_KtAkO4uHS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ea5ca62a437aa570476e29659bb25cdf846a2b95/330-ci-beans/src/test/java/io/harness/CiBeansComponentTest.java#L36)):*
<details><summary><i>Previously, hemanthmantri (Hemanth Mantri) wrote…</i></summary><blockquote>

Please add a comment explaining what this is doing. In the example please also indicate how to specify a list of these dependent classes.
</blockquote></details>

Done.

**Review by rutvijmehta-harness — APPROVED** [review]
> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: 2. unresolved discussions.

**Review by rutvijmehta-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: 2. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: 1. unresolved discussions.

**Review by hemanthmantri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32737)* status: :shipit: complete! ✓

___
*[`999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java` line 14 at r1](https://reviewable.io/reviews/harness/harness-core/32737#-N19c8jT0CYaHrIdDxzE-r1-14:-N1KVtd_AAg9yHJ_Iqqb:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/2c158f22d4b9dafc13f3602da9a373d178fb5532/999-annotations/src/main/java/io/harness/agent/sdk/HarnessHierarchy.java#L14)):*
<details><summary><i>Previously, jamie-harness (Jamie Li) wrote…</i></summary><blockquote>

Let's do that in the doc. Will need a few more lines than comment
</blockquote></details>

OK

**Pull request merged**